src/libs/freetype/lv_freetype_outline.c

```diff
@@ -354,6 +360,40 @@ static lv_freetype_outline_t outline_create(
         .delta = 0
     };
 
+#if LV_USE_NEMA_GFX
+#if LV_USE_NEMA_VG
+    /*Calculate Total Segmenets Before decompose */
+    int tag_size = face->glyph->outline.n_points;
+    unsigned char * tags = face->glyph->outline.tags;
+    int segments = 0;
+    int vectors = 0;
+
+    for(int j = 0; j < tag_size; j++) {
+        if(tags[j] & 0x1 == 0x1) {
+            segments++;
+            vectors++;
+        }
+        else {
+            int jj = j + 1 < tag_size ? j + 1 : 0;
+            if(tags[jj] & 0x1) {
+                vectors++;
+            }
+            else {
+                segments++;
+                vectors += 2;
+            }
+        }
+    }
+    /*Also for every contour we may have a line for close*/
+    segments += face->glyph->outline.n_contours;
+    vectors += face->glyph->outline.n_contours;
+
+    lv_nema_gfx_path_t * nema_gfx_path = (lv_nema_gfx_path_t *) param.outline;
+    nema_gfx_path->data_size = vectors * 2;
```

Comment:
We should avoid doing it in common code. How about still relying on lv_array to dynamically extend the data_size?

Reply:
Dynamic allocation in those cases could cause major performance issues. It is quite common to have glyphs described with ~ 150 segments. That could cause 150 reallocations for segments points + 150 reallocations for data points and thats for one letter alone, in each frame (especially if the glyph is not cached that could cause a bigger issue). This block inside lv_freetype_outline.c is only enabled when LV_USE_NEMA_GFX is used, so if this is not a hard request from your side we would like to keep this solution.

Reply:
We can pre-alloc the array to hold 150 elements and doubles it whenever runs out of space.

If it's really critical, then dynamic allocation should be avoided.

> if this is not a hard request from your side we would like to keep this solution.

If it's really needed, we can add EVENT etc for it. Register the event in `src/draw/nema_gfx/lv_draw_nema_gfx_label.c` to handle it.

Reply:
I think @XuNeo is referring to [lv_array_resize](https://github.com/lvgl/lvgl/blob/master/src/misc/lv_array.h#L73C6-L73C21) which pre-allocates space for N items.

Reply:
@XuNeo I will send a new event with the lengths of data points and segments, in that way we will avoid any reference to Nema specific structures, also anyone who wants can handle the event and get the data. I will update the code soon.

Reply:
**Note:** As @FASTSHIFT suggested I will add the data needed in lv_freetype_outline_event_param_t structure, so that the length information can be read in LV_EVENT_CREATE